stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -228,21 +261,67 @@ extension Task {
     /// - SeeAlso: `Task.checkCancelled`
     /// - SeeAlso: `Task.isCancelled`
     public mutating func cancelAll() {
-      _taskCancel(self.task) // TODO: do we also have to go over all child tasks and cancel there?
+      _taskGroupCancelAll(task: _task, group: _group)
+    }
+
+    // Returns `true` if the group was cancelled, e.g. by `cancelAll`.
+    public var isCancelled: Bool {
+      return _taskIsCancelled(_task) ||
+        _taskGroupIsCancelled(task: _task, group: _group)
```

Comment:
Thanks to this we don't have to register any cancellation handlers on the task to trigger `cancelAll` 💡 

The end results are exactly the same:

- if parent is cancelled, group is cancelled, and so are all children of the parent -- including US and all our children; the group only signals that it's cancelled here so that the `add` stops accepting work.

I think this is a good impl since we save registering any callbacks / cancellation handlers

Reply:
We also avoid the thread local lookups to get the task, since we "have it" already and it is guaranteed not to change which task we're in. So this check is two atomic fetch relaxed operations